besu/src/main/java/org/hyperledger/besu/services/BesuPluginContextImpl.java

```diff
@@ -211,4 +212,10 @@ private static URL pathToURIOrNull(final Path p) {
 
     return Optional.empty();
   }
+
+  public Map<String, BesuPlugin> getNamedPlugins() {
+    return plugins.stream()
+        .filter(plugin -> plugin.getName().isPresent())
+        .collect(Collectors.toMap(plugin -> plugin.getName().get(), plugin -> plugin, (a, b) -> b));
```

Comment:
I think `.flatMap(plugin.getName().stream())` does what you want here

Reply:
Oh, maybe not, I see you still need the plugin for the map

Reply:
Indeed.